### PR TITLE
feat: auth api

### PR DESCRIPTION
**Pull request by ArjunGTX**
- Integrated authentication

### CODE REVIEW

**Review by sharathnair9999 — APPROVED** [review]
Keeping all the relevant pieces into different files looks good

**Review by tanishk15sharma — APPROVED** [review]
I liked that you are using useEffect for setting the `key` `values` in auth `useState`

**Comment by Nikhil-Nirmal** [comment]
code looks good to merge

**Review by Nikhil-Nirmal — APPROVED** [review]

File: src/contexts/AuthContext.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  const [auth, setAuth] = useState({
+    userId: localStorage.getItem("userId"),
+    isLoggedIn: JSON.parse(localStorage.getItem("isLoggedIn")),
+    encodedToken: localStorage.getItem("encodedToken"),
+  });
+
+  useEffect(() => {
+    localStorage.setItem("userId", auth.userId);
+    localStorage.setItem("isLoggedIn", auth.isLoggedIn);
```
Comment (rehanmohiuddin) [inline src/contexts/AuthContext.jsx:19]: Store only one json stringified object in localstorage as user keep isLoggedIn in reducer state just a suggestion

Comment (ArjunGTX) [reply src/contexts/AuthContext.jsx:19]: Did to this way for easy access and to keep each item independent of each other

**Pull request merged**